File: src/examples/tutorial/ascent_intro/python/conduit_example3.py
###############################################################################
# Copyright (c) Lawrence Livermore National Security, LLC and other Ascent
# Project developers. See top-level LICENSE AND COPYRIGHT files for dates and
# other details. No copyright assignment is required to contribute to Ascent.
###############################################################################

import conduit
import numpy as np

#
# Conduit's Node trees hold strings or bitwidth style numeric array leaves
#
# In C++: you can pass raw pointers to numeric arrays or 
#  std::vectors with numeric values
# 
# In python: numpy ndarrays are used for arrays
#
n = conduit.Node()
a = np.zeros(10, dtype=np.int32)

a[0] = 0
a[1] = 1
for i in range(2,10):
   a[i] = a[i-2] + a[i-1]

n["fib"].set(a);
print(n.to_yaml());

File: src/examples/tutorial/ascent_intro/cpp/conduit_example3.cpp
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
// Copyright (c) Lawrence Livermore National Security, LLC and other Ascent
// Project developers. See top-level LICENSE AND COPYRIGHT files for dates and
// other details. No copyright assignment is required to contribute to Ascent.
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//

//-----------------------------------------------------------------------------
///
/// file: conduit_example3.cpp
///
//-----------------------------------------------------------------------------

#include <iostream>
#include <vector>
#include "conduit.hpp"

using namespace conduit;

int main()
{
    //
    // Conduit's Node trees hold strings or bitwidth style numeric array leaves
    //
    // In C++: you can pass raw pointers to numeric arrays or
    // std::vectors with numeric values
    //
    // In python: numpy ndarrays are used for arrays
    //

    Node n;
    int *A = new int[10];
    A[0] = 0;
    A[1] = 1;
    for (int i = 2 ; i < 10 ; i++)
        A[i] = A[i-2]+A[i-1];
    n["fib"].set(A, 10);

    std::cout << n.to_yaml() << std::endl;

    delete [] A;
}

